### PR TITLE
Lex v2 integration

### PR DESCRIPTION
**Pull request by tehzwen**
This is a big one! If everyone can please have look at it I'd appreciate it!
- Added Lex V2 Resources
- Removed V1 Unused resources
- Added tests
- Updated documentation
- Project management changes etc

### CODE REVIEW

**Review by jessedoyle — DISMISSED** [review]
🚀

File: __fixtures__/v2/events/bot/update.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "RequestId": "cd9c932d-65d6-4ed4-a56d-3347f8144e49",
+  "LogicalResourceId": "MyBot",
+  "PhysicalResourceId": "BOT_ID",
+  "ResourceType": "AWS::CloudFormation::CustomResource",
+  "ResourceProperties": {
+    "ServiceToken": "arn:aws:lambda:us-east-1:123456789000:function:CustomResourceStackTestZA-v2LexBotCustomResourcesS-lAXUQkcY7EVR",
+    "props": "{\"botTags\":{\"Deployer\":\"test@example.com\"},\"dataPrivacy\":{\"childDirected\":false},\"idleSessionTTLInSeconds\":100,\"roleArn\":\"arn:aws:iam::123456789000:role/LexCustomResourcesStack-v2LexCustomLambdaRoleD6144-17UINKA2H29EB\",\"description\":\"V2_Test_Bot_CDK\",\"botName\":\"V2_Test_Bot\"}"
+  },
+  "OldResourceProperties": {
+    "ServiceToken": "arn:aws:lambda:us-east-1:157153201295:function:CustomResourceStackTestZA-v2LexBotCustomResourcesS-lAXUQkcY7EVR",
```
Comment (mvandenbeuken) [inline __fixtures__/v2/events/bot/update.json]: is this one a real account ID? We may want to do a historical git rewrite if it is.

File: __fixtures__/v2/events/intent/create.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        "props": "{\"botId\":\"E2MGJRZBQA\",\"intentName\":\"TireService\",\"localeId\":\"en_US\",\"botVersion\":\"DRAFT\",\"sampleUtterances\":[{\"utterance\":\"My car has a flat tire\"},{\"utterance\":\"My truck has a flat tire\"},{\"utterance\":\"My van has a flat tire\"},{\"utterance\":\"I need a flat tire replaced\"},{\"utterance\":\"I need a flat tire replacement\"},{\"utterance\":\"I need you to replace my flat tire\"},{\"utterance\":\"I need you to replace my tire\"},{\"utterance\":\"I have a flat tire\"},{\"utterance\":\"I'd like a flat tire replacement\"},{\"utterance\":\"I'd like my flat tire replaced\"},{\"utterance\":\"I'd like you to replace my flat tire\"},{\"utterance\":\"I've got a flat\"},{\"utterance\":\"Can you replace my flat tire\"},{\"utterance\":\"A tire is flat\"},{\"utterance\":\"Replace my flat tire\"},{\"utterance\":\"Replace a flat tire\"},{\"utterance\":\"No air in tire\"},{\"utterance\":\"Flat tire\"},{\"utterance\":\"Flat tire service\"},{\"utterance\":\"Tire service\"}],\"dialogCodeHook\":{\"enabled\":true},\"fulfillmentCodeHook\":{\"enabled\":true},\"description\":\"An intent for servicing tires!\"}"
```
Comment (mvandenbeuken) [inline __fixtures__/v2/events/intent/create.json]: do we want to obfuscate some of the intents/slots/utterances in our fixtures to make them generic?

File: src/v2/handlers/lex-bot-alias/package.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "v2-lex-bot-alias-handler",
```
Comment (mvandenbeuken) [inline src/v2/handlers/lex-bot-alias/package.json:2]: are we thinking of using yarn across the board? Do we want to do that now and remove the package files/add them to gitignore?

File: src/v2/handlers/lex-bot-locale/index.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        botVersion: params.botVersion || "DRAFT",
+        localeId: event.PhysicalResourceId
+      });
+      const response = await client.send(deleteCommand);
+
+      return {
+        PhysicalResourceId: response.localeId
+      };
+    }
+    case "Update": { // to be adjusted
```
Comment (mvandenbeuken) [inline src/v2/handlers/lex-bot-locale/index.js]: is the comment still valid?

File: src/v2/handlers/lex-intent/index.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  region: process.env.REGION || "us-east-1",
+  logger
+});
+
+
+const handler = async (event, context) => {
+  try {
+    logger.info(event);
+    const params = JSON.parse(event.ResourceProperties.props);
+    let response = {};
```
Comment (mvandenbeuken) [inline src/v2/handlers/lex-intent/index.js]: do we want to use the same pattern being used in other handlers with scoping on the switch and const response = in the blocks? I think you're fine, but maybe just for the sake of consistency.

**Review by mvandenbeuken — CHANGES_REQUESTED** [review]
Looking really good team. Just a few comments.

File: src/v2/handlers/lex-bot-alias/package.json (hunk 3 above)
Comment (jessedoyle) [reply src/v2/handlers/lex-bot-alias/package.json:2]: Good catch here - the lambda handlers should have `yarn.lock` over `package-lock.json`.

File: __fixtures__/v2/events/bot/update.json (hunk 1 above)
Comment (jessedoyle) [reply __fixtures__/v2/events/bot/update.json]: History is rewritten!

File: __fixtures__/v2/events/intent/create.json (hunk 2 above)
Comment (jessedoyle) [reply __fixtures__/v2/events/intent/create.json]: Yup, done!

File: src/v2/handlers/lex-bot-locale/index.js (hunk 4 above)
Comment (jessedoyle) [reply src/v2/handlers/lex-bot-locale/index.js]: Removed

File: src/v2/handlers/lex-intent/index.js (hunk 5 above)
Comment (jessedoyle) [reply src/v2/handlers/lex-intent/index.js]: Yes, definitely!

**Review by mvandenbeuken — APPROVED** [review]
Great work all! Approved!

**Pull request merged**